### PR TITLE
Reader move tab layout to parent part 1

### PR DESCRIPTION
**Pull request by malinajirka**
Partially fixes https://github.com/wordpress-mobile/WordPress-Android/issues/11849
This is a first PR in a series of PRs which is trying to introduce a regular TabLayout+ViewPager2 in the main Reader view. The current implementation uses FilteredRecyclerView component (which is not a RecyclerView :P)  that has some downsides (eg. swipe gesture doesn't work).

To test:
- You can test that the parent TabLayout works as expected and the nested tablayout is opened on the tab that corresponds to the selected tab in the parent.
Removing the nested tablayout etc will be done in future PRs.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/layout/reader_parent_post_list_fragment.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <com.google.android.material.tabs.TabLayout
+        android:id="@+id/tab_layout"
```
Comment (planarvoid) [inline WordPress/src/main/res/layout/reader_parent_post_list_fragment.xml:8]: I think there is a flag that stops the tabs from having constant width, now they look like this:
Screenshot_1588852773

**Review by planarvoid — COMMENTED** [review]
There is a bug that after a while when randomly clicking on the tabs, the Following tab contains content from the Discover tab (and you never see the Following tab after that point). I'm sending the GIF in a DM

File: WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderParentPostListViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import kotlinx.coroutines.launch
+import org.wordpress.android.models.ReaderTagList
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.reader.usecases.LoadReaderTabsUseCase
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ScopedViewModel
+import javax.inject.Inject
+import javax.inject.Named
+
+class ReaderParentPostListViewModel @Inject constructor(
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderParentPostListViewModel.kt]: 👍 Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/11840/commits/c4a59aea6cccb39f1974a07f43b28c0b59b03c5e

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.models.ReaderTagList
+import org.wordpress.android.ui.reader.ReaderTypes.ReaderPostListType
+import org.wordpress.android.ui.reader.viewmodels.ReaderParentPostListViewModel
+import javax.inject.Inject
+
+class ReaderParentPostListFragment : Fragment(R.layout.reader_parent_post_list_fragment) {
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt]: 👍 Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/11840/commits/c4a59aea6cccb39f1974a07f43b28c0b59b03c5e

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    private fun initViewModel(view: View) {
+        viewModel = ViewModelProviders.of(this, viewModelFactory).get(ReaderParentPostListViewModel::class.java)
+        startObserving(view)
+        viewModel.start()
+    }
+
+    private fun startObserving(view: View) {
+        viewModel.tabs.observe(viewLifecycleOwner, Observer { event ->
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt]: 👍 Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/11840/commits/4c79e54c06a0ea3dedd6783f3adef204b04c67c1

File: WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderParentPostListViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    private val loadReaderTabsUseCase: LoadReaderTabsUseCase
+) : ScopedViewModel(mainDispatcher) {
+    private var started: Boolean = false
+    private val _tabs = MutableLiveData<ReaderTagList>()
+    val tabs: LiveData<ReaderTagList> = _tabs
+
+    fun start() {
+        if (started) return
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderParentPostListViewModel.kt]: Yeah, I don't like the `started` flag either. However, I would still keep it unless we refactor all the places through out the app.

File: WordPress/src/main/res/layout/reader_parent_post_list_fragment.xml (hunk 1 above)
Comment (malinajirka) [reply WordPress/src/main/res/layout/reader_parent_post_list_fragment.xml]: Good catch! Fixed in f8338b915a0e8b0d547bf24c8cd7f67e5ebe4178

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    private fun initViewPager(tags: ReaderTagList, view: View) {
+        val adapter = TabsAdapter(this, tags)
+        val viewPager = view.findViewById<ViewPager2>(R.id.view_pager)
+        viewPager.adapter = adapter
+
+        val tabLayout = view.findViewById<TabLayout>(R.id.tab_layout)
+        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
+            tab.text = tags[position].tagDisplayName // TODO should we use displayname or title?
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt]: I'm not sure it's exactly what you meant, but I introduced a UI state in fc2e41aa0b890bb4ab126545c922942dff6544fb

**Comment by malinajirka** [comment]
Thanks @planarvoid ! I've replied to your comments and fixed the issues. 

>There is a bug that after a while when randomly clicking on the tabs, the Following tab contains content from the Discover tab (and you never see the Following tab after that point). I'm sending the GIF in a DM

This issue will be fixed in the follow-up PRs.


It's ready for another round, thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderParentPostListFragment.kt]: that's almost exactly what I meant 👍.

**Review by planarvoid — APPROVED** [review]
Thanks for the changes 👍 looks good :shipit:

**Pull request merged**